app/controllers/api/v1/players_controller.rb

```diff
@@ -4,20 +4,45 @@ class Api::V1::PlayersController < ApplicationController
   def create
     if game.present?
       game.players.create(user_id: current_user.id)
+      game.players.alive!
       render json: game
     else
       flash[:alert] = 'You has errors!'
       render json: { status: :conflict }
     end
   end
 
+  def death_confirm
+    player.death_confirm!
+  end
+
+  def player_killed
+    target_info.dead! if target_info.death_confirm?
+    available_target ||= players.alive_users.all_except(current_user.id).sample
+    return current_user.current_game.finished! if available_target.nil?
+    player.update(target_id: available_target.id)
+    player.target_ids.push(available_target.id)
+  end
+
+  def error_death
+    target_info.alive! unless target_info.dead
+  end
+
   def destroy
     player.destroy
     render json: {}, status: :accepted
   end
 
   private
 
+  def target_info
+    current_user.current_game.players.find(player.target_id)
```

Comment:
change to `player.target`

And add association on Player
`belongs_to :target, class_name: "Player"`